tick/optim/model/poisreg.py

```diff
@@ -169,3 +170,48 @@ def _hessian_norm(self, coeffs: np.ndarray,
             raise ValueError("``link`` must be either 'exponential' or "
                              "'linear'.")
 
+    def _sdca_primal_dual_relation(self, l_l2sq, dual_vector):
+        # In order to solve the same problem than other solvers, we need to
+        # rescale the penalty parameter if some observations are not
+        # considered in SDCA. This is useful for Poisson regression with
+        # identity link
+        if self.link == "identity":
+            scaled_l_l2sq = l_l2sq * self.n_samples / self._sdca_rand_max
+        else:
+            scaled_l_l2sq = l_l2sq
+
+        primal_vector = np.empty(self.n_coeffs)
+        self._model.sdca_primal_dual_relation(scaled_l_l2sq, dual_vector,
+                                              primal_vector)
+        return primal_vector
+
+    @property
+    def _sdca_rand_max(self):
+        if self.link == "identity":
+            non_zero_labels = self.labels != 0
+            return non_zero_labels.sum().item()
+        else:
+            raise NotImplementedError()
+
+    def dual_loss(self, dual_coeffs):
```

Comment:
Yes, I don't think the efficiency is important here as this is only used to verbose things. However, I understand that tick might be rather seen as a Python wrapper over a C++ API. Maybe should we create ModelSDCAFriendly in another task though as this task is already pretty big.